3.preprocessing_features/scripts/2.single_cell_processing.py

```diff
@@ -12,9 +12,7 @@
 import pprint
 
 import pandas as pd
-
-from pycytominer import annotate, normalize, feature_select
-
+from pycytominer import annotate, feature_select, normalize
```

Comment:
I used a tool called [nbqa](https://github.com/nbQA-dev/nbQA), which stands for Notebook Quality Assurance (a name I find pretty amusing, to be honest). It’s a powerful utility that allows Python code quality and formatting tools to work seamlessly with Jupyter notebooks (*.ipynb). Essentially, it extracts the Python code from a notebook’s code cells, applies tools like  ruff for linting and formatting, or mypy for type-checking, and then reinserts the processed code back into the notebook, all while preserving its structure.

Since I didn't use pre-commit to do, I basically used it in the terminal ike this:

If you want to use nbqa for both linting and formatting your Jupyter notebooks with ruff, you can configure ruff to handle both tasks. ruff supports a range of features, including linting (catching style issues and potential errors) and formatting (enforcing consistent code style, similar to black).

Here’s how you can set up nbqa to use ruff for both:
1. Install the Required Tools:
```bash
pip install nbqa ruff
```

2. Run Ruff with nbqa:
```bash
nbqa ruff --fix your_notebook.ipynb
```

- The `--fix` option tells ruff to apply fixes, which includes both linting corrections and formatting changes.

I hope this answers your question!